kura/org.eclipse.kura.linux.clock/src/main/java/org/eclipse/kura/linux/clock/AbstractNtpClockSyncProvider.java

```diff
@@ -42,12 +39,19 @@
     protected boolean isSynced;
     protected int syncCount;
 
+    private ScheduledFuture<?> future;
+
     @Override
-    public void init(Map<String, Object> properties, ClockSyncListener listener) throws KuraException {
-        this.properties = properties;
+    public void init(ClockServiceConfig clockServiceConfig, ScheduledExecutorService scheduler,
+            ClockSyncListener listener) throws KuraException {
+        this.scheduler = scheduler;
         this.listener = listener;
-
-        readProperties();
+        this.ntpHost = clockServiceConfig.getNtpHost();
```

Comment:
Since `ClockServiceConfig` is immutable, we could also define a `clockServiceConfig` field for storing it instead of the multiple per property fields, but it is fine also as is.

Reply:
I don't think we gain so much if we keep a single field. Maybe it is more clear, being an abstract class to have the fields separate. Don't know, I think that both solutions work fine.